File: external/HDF-EOS/hdfeos/samples/appendfield.f

c
c  In this example we will write to an appendable field.
c


      program appendfield

      integer            status, i, swfldinfo, swdetach, swclose
      integer            swwrfld, swrdfld
      integer*2          inarray(5), outarray(20)
      integer*4          swfid, swid, rank, dims(1), ntype
      integer*4          start(1), stride(1), count(1)
      integer*4          swopen, swattach
      character*72       dimlist
      
      integer*4          DFACC_RDWR
      parameter (DFACC_RDWR=3)
      
      
      do i=1,5
         inarray(i) = i
      enddo
      
      swfid = swopen("SwathFile.hdf", DFACC_RDWR)
      swid = swattach(swfid, "Swath1")


c     Write 5 records to field
c     ------------------------
      start(1) = 0
      stride(1) = 1
      count(1) = 5
      status = swwrfld(swid, "Count", start, stride, count, inarray)

      status = swfldinfo(swid, "Count", rank, dims, ntype, dimlist)
      write(*,*) 'Number of elements after first write:' ,dims(1)



c     Append 1 record to field
c     ------------------------
      start(1) = dims(1)
      count(1) = 1
      status = swwrfld(swid, "Count", start, stride, count, inarray)
      status = swfldinfo(swid, "Count", rank, dims, ntype, dimlist)
      write(*,*) 'Number of elements after append:' ,dims(1)

      start(1) = 0
      count(1) = dims(1)
      status = swrdfld(swid, "Count", start, stride, dims, outarray)    
      do i=1,dims(1)
         write(*,*) 'Data Element:', outarray(i)
      enddo

      status = swdetach(swid)
      status = swclose(swfid)

      stop
      end

File: external/HDF-EOS/hdfeos/samples/AppendField.c
#include "hdf.h"

/*
 * In this example we will demonstrate the use of the unlimited dimension
 * in creating a field that can be appended to.
 */


main()
{

    intn            status, i;
    int16           inarray[5] = {1, 2, 3, 4, 5}, outarray[10];
    int32           swfid, SWid, rank, dims[1], ntype;
    int32           start[1], stride[1], count[1];
    

    swfid = SWopen("SwathFile.hdf", DFACC_RDWR);
    SWid = SWattach(swfid, "Swath1");

    status = SWfieldinfo(SWid, "Count", &rank, dims, &ntype, NULL);
    printf("Initial number of elements: %d\n",dims[0]);

    /* Write 5 records to field */
    /* ------------------------ */
    count[0] = 5;
    SWwritefield(SWid, "Count", NULL, NULL, count, inarray);


    status = SWfieldinfo(SWid, "Count", &rank, dims, &ntype, NULL);
    printf("Number of elements after first write: %d\n",dims[0]);


    
    /* Append 1 record to field */
    /* ------------------------ */
    start[0] = dims[0];
    count[0] = 1;
    SWwritefield(SWid, "Count", start, NULL, count, inarray);


    status = SWfieldinfo(SWid, "Count", &rank, dims, &ntype, NULL);
    printf("Number of elements after append: %d\n",dims[0]);

    SWreadfield(SWid, "Count", NULL, NULL, dims, outarray);
    for (i=0; i<dims[0]; i++) printf("Data Element: %d\n",outarray[i]);
    
    status = SWdetach(SWid);
    status = SWclose(swfid);

    return;
}
